### DISCUSSION TITLE
Setting up mock Plex server

### DISCUSSION THREAD

**Discussion by najmsheikh** [itsToggle/plex_debrid#42] [General]
Hey @itsToggle ! Thank you so much for setting this up! In your README you mentioned the side project that you're working on. Would you be able to give a more detailed high level overview of how you get that all working? My understanding of Plex Discover was that its limited to a specific set of providers. But I imagine you got around that by using a mocked Plex server as the content provider? What about matching for content that hasn't been synced on the personal debrid account yet?

**Comment by itsToggle** [comment]
Sure! You are right, I'm mocking a plex server to add content providers. 

Whenever you open content inside plex discover, an availability request to all servers registered under your account is made to check if you already own the content. So to inject information, we first need to register a server under our account and then respond to these availability requests in an altered way. 

Since I haven't figured out which requests need to be made and responded to in order to register a new server, I'm simply using a second pc for this step.

1. Install Plex media server on a second PC.
2. Claim the server with your account.
3. The server is now registered under your account and availability requests will be made to the servers address whenever you open a movie/show inside plex discover. 

To respond to the availability requests in an altered way, we need plex to redirect the availability requests to some other address which we can listen to with an http server:

4. Run a python https server that can responds to the availability requests. (There are also some more requests this "mock" server needs to be able to respond to, in order for plex to accept this as a content provider - mainly the /providers requests)
5. The python server needs to run on a publicly reachable url that can be secured by https (this can be done with ngrok for free, or if one owns a vps)
6. Point the plex server to the python https "mock" server url, by adding a "custom server access url".
7. Turn off the plex server, so that all requests will be made to the python server

And that's it. You can now respond to availability requests in any way you like. The way I did it in the video was that as soon as an availability request for a media item would be recieved, I would scrape releases for this movie/show/season/episode, check debrid services for cached releases and then return the available streaming qualities. When one of those faked streaming options is clicked, I can download the selected release to a non-"mock" server and refresh the library.

Since this happens entirely over https, all plex clients work with any features you write for this mock server.

I hope this helps :) I would be very interested in any project that might come from this and will gladly offer my help :)
